plugins/modules/user.py

```diff
@@ -0,0 +1,261 @@
+    role_uuids = []
+    for role_name in module.params["roles"]:
+        role = Role.get_role_from_name(role_name, rest_client, must_exist=True)
+        role_uuids.append(role.uuid)
+    return role_uuids
+
+
+def data_for_create_user(module, rest_client: RestClient):
+    data = {}
+    data["username"] = module.params[
+        "username"
+    ]  # For api only username is required - SHOULD WE MAKE OTHER PARAMETERS REQUIRED?
+    if module.params["password"]:
+        data["password"] = module.params["password"]
+    if module.params["full_name"]:
+        data["fullName"] = module.params["full_name"]
+    if module.params["roles"]:
+        data["roleUUIDs"] = get_role_uuids(module, rest_client)
+    if module.params["session_limit"] is not None:
+        data["sessionLimit"] = module.params["session_limit"]
+    return data
+
+
+def create_user(module, rest_client: RestClient):
+    data = data_for_create_user(module, rest_client)
+    user = User.create(rest_client, data).to_ansible(rest_client)
+    return (
+        True,
+        user,
+        dict(before={}, after=user),
+    )
+
+
+def data_for_update_user(module, rest_client: RestClient, user):
+    data = {}
+    if module.params["new_username"]:
+        if user.username != module.params["new_username"]:
+            data["username"] = module.params["new_username"]
+    if module.params["password"]:
+        data["password"] = module.params["password"]  # password isn't visible in GET
+    if module.params["full_name"]:
+        if user.full_name != module.params["full_name"]:
+            data["fullName"] = module.params["full_name"]
+    if module.params["roles"]:
+        role_uuids = get_role_uuids(module, rest_client)
+        if user.role_uuids != role_uuids:
+            data["roleUUIDs"] = role_uuids
+    if (
+        module.params["session_limit"] is not None
+    ):  # "is not None" needed to be able to write zero
+        if user.session_limit != module.params["session_limit"]:
+            data["sessionLimit"] = module.params["session_limit"]
+    return data
+
+
+def update_user(module, rest_client: RestClient, user):
+    data = data_for_update_user(module, rest_client, user)
+    if data:
+        user.update(rest_client, data)
+        user_after = User.get_user_from_uuid(
+            user.uuid, rest_client, must_exist=True
+        ).to_ansible(rest_client)
+        user = user.to_ansible(rest_client)
+        return (
+            True,
+            user_after,
+            dict(before=user, after=user_after),
+        )
+    user = user.to_ansible(rest_client)
+    return (
+        False,
+        user,
+        dict(before=user, after=user),
+    )
+
+
+def delete_user(rest_client: RestClient, user):
+    if user:
+        user.delete(rest_client)
+        return True, dict(), dict(before=user.to_ansible(rest_client), after={})
+    return False, dict(), dict(before={}, after={})
```

Comment:
I almost comment why you return `dict(before={}, ` (line 203 - last one) if you deleted a user?

A bit easier to read if special case of user not present is hadled first, the the normal code excution does not need to be in the `if`.

```
    if not user:
        return False, dict(), dict(before={}, after={})
    user.delete(rest_client)
    return True, dict(), dict(before=user.to_ansible(rest_client), after={})
```

Reply:
Changed the code order as suggested